apps/server/src/modules/task/service/submission.service.ts

```diff
@@ -26,4 +31,104 @@ export class SubmissionService {
 
 		await this.submissionRepo.delete(submission);
 	}
+
+	async deleteSubmissionsByUserId(userId: EntityId): Promise<DomainOperation> {
+		this.logger.info(
+			new DataDeletionDomainOperationLoggable(
+				'Deleting data from Submissions',
+				DomainName.SUBMISSIONS,
+				userId,
+				StatusModel.PENDING
+			)
+		);
+		const submissions = await this.submissionRepo.findAllByUserId(userId);
+		const submissionsToDelete: Submission[] = [];
+
+		if (submissions[1] > 0) {
+			submissions[0].forEach((submission) => {
+				if (submission.courseGroup === null && submission.teamMembers.length === 1) {
+					submissionsToDelete.push(submission);
+				}
+			});
+		}
+
+		const submissionsToDeleteCount = submissionsToDelete.length;
+		if (submissionsToDeleteCount > 0) {
+			await this.submissionRepo.delete(submissionsToDelete);
+		}
+
+		const result = DomainOperationBuilder.build(
+			DomainName.SUBMISSIONS,
+			OperationType.DELETE,
+			submissionsToDelete.length,
+			this.getSubmissionsId(submissionsToDelete)
+		);
+
+		this.logger.info(
+			new DataDeletionDomainOperationLoggable(
+				'Successfully deleted data from Submissions',
+				DomainName.SUBMISSIONS,
+				userId,
+				StatusModel.FINISHED,
+				submissionsToDeleteCount,
+				0
+			)
+		);
+
+		return result;
+	}
+
+	async updateSubmissionByUserId(userId: EntityId): Promise<DomainOperation> {
```

Comment:
Also in case of this method - without looking into the method's code it's not clear what's the point of the update operation here and why you need a userId for that. I would suggest something like `removeUserReferencesFromSubmissions`.